creator-node/src/services/stateMachineManager/stateMachineUtils.js

```diff
@@ -134,47 +134,34 @@ const retrieveClockValueForUserFromReplica = async (replica, wallet) => {
 /**
  * Returns an object that can be returned from any state machine job to record a histogram metric being observed.
  * Example: to call histogram.observe('response_time', { code: '200' }, 1000), you would call this function with:
- * makeHistogramToRecord('response_time', 1000, 'code', '200')
- * @param {string} metricName the key of the name of the metric from prometheus.constants
+ * makeHistogramToRecord('response_time', 1000, { code: '200' })
+ * @param {string} metricName the name of the metric from prometheus.constants
  * @param {number} metricValue the value to observe
- * @param {string} [metricLabelName] the optional name of the label
- * @param {string} [metricLabelValue] the optional value of the label
+ * @param {string} [metricLabels] the optional mapping of metric label name => metric label value
  */
-const makeHistogramToRecord = (
-  metricName,
-  metricValue,
-  metricLabelName = '',
-  metricLabelValue = ''
-) => {
-  if (!Object.keys(MetricNames).includes(metricName)) {
-    throw new Error(
-      `Invalid metricName: ${metricName}. Options: ${JSON.stringify(
-        Object.keys(MetricNames)
-      )}`
-    )
+const makeHistogramToRecord = (metricName, metricValue, metricLabels = {}) => {
+  if (!Object.values(MetricNames).includes(metricName)) {
+    throw new Error(`Invalid metricName: ${metricName}`)
   }
   if (typeof metricValue !== 'number') {
     throw new Error(`Invalid non-numerical metricValue: ${metricValue}`)
   }
-  const useMetricLabel = metricLabelName && metricLabelValue
-  if (
-    useMetricLabel &&
-    !Object.keys(MetricLabels[metricName])?.includes(metricLabelName)
-  ) {
-    throw new Error(`Invalid metricLabelName: ${metricLabelName}`)
-  }
-  if (
-    useMetricLabel &&
-    !MetricLabels[metricName][metricLabelName]?.includes(metricLabelValue)
-  ) {
-    throw new Error(`Invalid metricLabelValue: ${metricLabelValue}`)
+  const labelNames = Object.keys(MetricLabels[metricName])
+  for (const [labelName, labelValue] of Object.entries(metricLabels)) {
+    if (!labelNames?.includes(labelName)) {
+      throw new Error(`Metric label has invliad name: ${labelName}`)
+    }
+    const labelValues = MetricLabels[metricName][labelName]
+    if (!labelValues?.includes(labelValue)) {
+      throw new Error(`Metric label has invalid value: ${labelValue}`)
+    }
   }
 
   const metric = {
-    metricName: MetricNames[metricName],
-    metricType: MetricTypes.HISTOGRAM,
+    metricName,
+    metricType: 'HISTOGRAM',
```

Comment:
ah yeah, interesting - this should prob be a constant in prometheus.constants but also fine